alibuild_helpers/cmd.py

```diff
@@ -10,10 +10,9 @@
 
 def execute(command, printer=debug):
   popen = subprocess.Popen(command, shell=is_string(command), stdout=subprocess.PIPE)
-  lines_iterator = iter(popen.stdout.readline, "")
-  for line in lines_iterator:
+  for line in iter(popen.stdout.readline, ""):
     if not line: break
-    printer(to_unicode(line).strip("\n")) # yield line
+    printer("%s", to_unicode(line).strip("\n"))
```

Comment:
Why doing the formatting for "%s"?

Reply:
Because `line` can contain `%` signs (e.g. `[100%] Compiling xyz.cxx`), which confuses the string formatting.